pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -1556,8 +1563,31 @@ public void setCharacterStream(int parameterIndex, Reader value, long length)
     throw Driver.notImplemented(this.getClass(), "setCharacterStream(int, Reader, long)");
   }
 
+  private String getString(Reader value) throws SQLException {
+    try {
+      StringBuilder v = new StringBuilder(1024);
+      char[] buf = new char[1024];
+      int nRead = 0;
+      while (nRead > -1) {
+        nRead = value.read(buf);
+        if (nRead > 1) {
+          v.append(buf, 0, nRead);
+        }
+      }
+      return v.toString();
+    } catch (IOException ioe) {
+      throw new PSQLException(GT.tr("Provided Reader failed."), PSQLState.UNEXPECTED_ERROR, ioe);
+    }
+  }
+
   public void setCharacterStream(int parameterIndex, Reader value) throws SQLException {
```

Comment:
Sorry for the late review.

Technically speaking:
1) PostgreSQL protocol does not support streaming data of unknown length. That means, for "unknown size", the only way out is to buffer the value somehow.
2) "buffering the stream into `java.lang.String`" does not differ much from `org.postgresql.jdbc.PgPreparedStatement#setCharacterStream(int, java.io.Reader, int)`

Did you consider unifying existing `org.postgresql.jdbc.PgPreparedStatement#setCharacterStream(int, java.io.Reader, int)`, and other `setCharacterStream` methods?

It looks like your implementation brings yet another "save Reader into String" implementation with no code (and bug :) ) sharing with existing code around similar `setCharacterStream` methods.

PS. `StreamWrapper` does buffer the contents into `byte[]`. I think it is possible to keep stream all the way down and stream data to the network socket, however it is not implemented yet.